modules/recon/dns/dns.go

```diff
@@ -0,0 +1,344 @@
+package dns
+
+import (
+	"encoding/gob"
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/miekg/dns"
+	"github.com/netm4ul/netm4ul/core/config"
+	"github.com/netm4ul/netm4ul/core/server/database"
+	"github.com/netm4ul/netm4ul/modules"
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// DnsResult represent the parsed ouput
+type DnsResult struct {
+	Types    map[string][]string
+	resolver string
+	err      string
+}
+
+// ConfigToml : configuration model (from the toml file)
+type ConfigToml struct {
+	RandomDns bool `toml:"randomDNS"`
+}
+
+// Dns "class"
+type Dns struct {
+	// Config : exported config
+	Config ConfigToml
+}
+
+//NewDns generate a new Dns module (type modules.Module)
+func NewDns() modules.Module {
+	gob.Register(DnsResult{})
+	var d modules.Module
+	d = &Dns{}
+	return d
+}
+
+// Name : name getter
+func (D *Dns) Name() string {
+	return "Dns"
+}
+
+// Author : Author getter
+func (D *Dns) Author() string {
+	return "Rzbaa"
+}
+
+// Version : Version  getter
+func (D *Dns) Version() string {
+	return "0.1"
+}
+
+// DependsOn : Generate the dependencies requirement
+func (D *Dns) DependsOn() []modules.Condition {
+	var _ modules.Condition
+	return []modules.Condition{}
+}
+
+/*
+	Usefull command
+	curl -XPOST http://localhost:8080/api/v1/projects/FirstProject/run/dns
+	check db: Db.projects.find()
+	remove all data: db.projects.remove({})
+*/
+
+// Run : Main function of the module
+func (D *Dns) Run(data []string) (modules.Result, error) {
+
+	// Banner
+	fmt.Println("DNS world!")
+
+	// Generate config file
+	D.ParseConfig()
+
+	// Get fqdn of domain
+	domain := "edznux.fr"
+	fqdn := dns.Fqdn(domain)
+
+	// instanciate DnsResult
+	result := DnsResult{}
+
+	// Get DNS IP address from our config file
+	resolverConfig := make(map[int]*dns.ClientConfig)
+
+	resolverList := strings.Split(config.Config.DNS.Resolvers, ",")
+	for i := 0; i < len(resolverList); i++ {
+		// DNS gog library need resolv.conf entry (nameserver server_ip)
+		dnsEntry := "nameserver " + resolverList[i]
+		r := strings.NewReader(dnsEntry)
+		config, _ := dns.ClientConfigFromReader(r)
+		resolverConfig[i] = config
+	}
+
+	// Map Types for DnsResult{} treatment
+	result.Types = make(map[string][]string)
+
+	// For all Type in dns library
+
+	if D.Config.RandomDns == true {
+		// random DNS resolver
+		for _, index := range dns.TypeToString {
+			config := resolverConfig[rand.Intn(len(resolverConfig))]
+			requestRoutine(index, result, fqdn, config)
+		}
+	} else {
+		// Normal iteration of DNS resolvers
+		i := 0
+		for _, index := range dns.TypeToString {
```

Comment:
hm, why not for i, index := range dns.TypeToString ?
instead of inc i independantly

Reply:
for _, val := range 'chaine'
_ prend l'index et val prend la valeur du coup c'est bizarre de nommer ta deuxième variable "index", puisque le véritable index de "dns.TypeToString" va dans ta variable _

Reply:
accepte mon PR enculé

Reply:
C'est good

Reply:
with <3